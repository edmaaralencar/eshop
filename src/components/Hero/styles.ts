import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    margin: 0 auto;

    height: 60rem;
    width: 100%;
    max-width: ${theme.container};

    padding: 0 4rem;

    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 4rem;

    background-color: ${theme.colors.primary.dark};

    @media (max-width: 768px) {
      display: flex;
      flex-direction: column;
      gap: 2.4rem;
      padding: 3.2rem;
    }

    @media (max-width: 450px) {
      height: 67rem;
    }
  `}
`

export const Content = styled.div`
  flex: 1;
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;

    h1 {
      color: ${theme.colors.white};
      font-size: 4rem;
    }

    p {
      color: ${theme.colors.white};
      font-size: 1.8rem;
      font-weight: 400;
      line-height: 2.4rem;

      margin-top: 2.6rem;
      margin-bottom: 3.2rem;
    }
  `}
`

export const ImageWrapper = styled.div`
  flex: 1;
  position: relative;

  height: 40rem;

  @media (max-width: 768px) {
    width: 40rem;
  }

  @media (max-width: 450px) {
    width: 100%;
    height: 50rem;
  }
`

export const ButtonWrapper = styled.div`
  width: 20rem;
`
