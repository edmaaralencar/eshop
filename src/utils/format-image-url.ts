export function formatImageUrl(image: string) {
  return `/products/${image}`
}
