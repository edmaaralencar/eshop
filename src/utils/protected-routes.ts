import { GetServerSidePropsContext } from 'next'
// import { unstable_getServerSession } from 'next-auth'
import { getSession } from 'next-auth/react'
// import { options } from 'pages/api/auth/[...nextauth].api'

async function protectedRoutes(context: GetServerSidePropsContext) {
  // const session = await unstable_getServerSession(
  //   context.req,
  //   context.res,
  //   options
  // )

  // if (!session) {
  //   return {
  //     redirect: {
  //       destination: `/sign-in?callbackUrl=${context.resolvedUrl}`,
  //       permanent: false
  //     }
  //   }
  // }

  const session = await getSession(context)

  if (!session) {
    context.res.setHeader(
      'Location',
      `/sign-in?callbackUrl=${context.resolvedUrl}`
    )
    context.res.statusCode = 302
  }

  return session
}

export default protectedRoutes
