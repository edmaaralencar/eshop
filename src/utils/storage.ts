const APP_KEY = '@eshop'

export function getStorageItem(key: string) {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(`${APP_KEY}.${key}`)
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  return JSON.parse(data!)
}

export function setStorageItem(key: string, value: any[]) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(`${APP_KEY}.${key}`, data)
}

export function removeStorageItem(key: string) {
  if (typeof window === 'undefined') return

  return window.localStorage.removeItem(`${APP_KEY}.${key}`)
}
