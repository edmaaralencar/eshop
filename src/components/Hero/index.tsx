import Button from 'components/Button'
import Image from 'next/image'
import { FiArrowRight } from 'react-icons/fi'
import * as S from './styles'

function Hero() {
  return (
    <S.Wrapper>
      <S.Content>
        <h1>Compre os melhores sapatos do mercado</h1>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Saepe
          suscipit deleniti, placeat nobis, ut, porro dignissimos distinctio.
        </p>
        <S.ButtonWrapper>
          <Button
            variant="outlined"
            icon={<FiArrowRight size={22} color="#ffffff" />}
            size="medium"
            as="a"
            href=""
          >
            Comprar
          </Button>
        </S.ButtonWrapper>
      </S.Content>
      <S.ImageWrapper>
        <Image src="/produto.png" alt="" layout="fill" />
      </S.ImageWrapper>
    </S.Wrapper>
  )
}

export default Hero
