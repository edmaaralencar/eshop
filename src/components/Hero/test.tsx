import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test-helper'

import Hero from '.'

describe('<Hero />', () => {
  it('should render the heading and button', () => {
    const { container } = renderWithTheme(<Hero />)

    expect(
      screen.getByRole('heading', {
        name: /Compre os melhores sapatos do mercado/i
      })
    ).toBeInTheDocument()

    expect(screen.getByRole('link')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
