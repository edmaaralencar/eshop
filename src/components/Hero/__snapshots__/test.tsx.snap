// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render the heading and button 1`] = `
.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: 1px solid #FFFFFF;
  color: #FFFFFF;
  -webkit-transition: border-color 0.1s ease-out;
  transition: border-color 0.1s ease-out;
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  padding: 0 2rem;
  height: 5.2rem;
}

.c3 span {
  font-weight: 300;
}

.c3:disabled {
  cursor: not-allowed;
}

.c3:hover {
  border-color: #e6e6e6;
  color: #e6e6e6;
}

.c3:hover svg {
  color: #e6e6e6 !important;
}

.c3 span {
  font-size: 1.8rem;
}

.c3 svg {
  margin-left: 3.2rem;
  margin-top: 0.1rem;
}

.c0 {
  margin: 0 auto;
  height: 60rem;
  width: 100%;
  max-width: 117rem;
  padding: 0 4rem;
  display: grid;
  grid-template-columns: 1fr 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4rem;
  background-color: #6C6CFF;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 h1 {
  color: #FFFFFF;
  font-size: 4rem;
}

.c1 p {
  color: #FFFFFF;
  font-size: 1.8rem;
  font-weight: 400;
  line-height: 2.4rem;
  margin-top: 2.6rem;
  margin-bottom: 3.2rem;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  height: 40rem;
}

.c2 {
  width: 20rem;
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 2.4rem;
    padding: 3.2rem;
  }
}

@media (max-width:450px) {
  .c0 {
    height: 67rem;
  }
}

@media (max-width:768px) {
  .c4 {
    width: 40rem;
  }
}

@media (max-width:450px) {
  .c4 {
    width: 100%;
    height: 50rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <h1>
      Compre os melhores sapatos do mercado
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Saepe suscipit deleniti, placeat nobis, ut, porro dignissimos distinctio.
    </p>
    <div
      class="c2"
    >
      <a
        class="c3"
        href=""
      >
        <span>
          Comprar
        </span>
        <svg
          color="#ffffff"
          fill="none"
          height="22"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 24 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="5"
            x2="19"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 5 19 12 12 19"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    class="c4"
  >
    <img
      alt="product"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fproduto.png&w=3840&q=75"
      srcset="/_next/image?url=%2Fproduto.png&w=640&q=75 640w, /_next/image?url=%2Fproduto.png&w=750&q=75 750w, /_next/image?url=%2Fproduto.png&w=828&q=75 828w, /_next/image?url=%2Fproduto.png&w=1080&q=75 1080w, /_next/image?url=%2Fproduto.png&w=1200&q=75 1200w, /_next/image?url=%2Fproduto.png&w=1920&q=75 1920w, /_next/image?url=%2Fproduto.png&w=2048&q=75 2048w, /_next/image?url=%2Fproduto.png&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
</section>
`;
